api/v1alpha1/releaselink_webhook.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var releaselinklog = logf.Log.WithName("releaselink-resource")
+
+func (r *ReleaseLink) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+// +kubebuilder:webhook:path=/mutate-appstudio-redhat-com-v1alpha1-releaselink,mutating=true,failurePolicy=fail,sideEffects=None,groups=appstudio.redhat.com,resources=releaselinks,verbs=create;update,versions=v1alpha1,name=mreleaselink.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Defaulter = &ReleaseLink{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *ReleaseLink) Default() {
+
+	// TODO(user): fill in your defaulting logic.
```

Comment:
Could you empty this function? And actually... are you sure about webhook requiring defaulting to work? It's a bit weird that we are forced to have them. 36 to 44 could go out probably.

Reply:
```api/v1alpha1/releaselink_webhook.go:38:5: cannot use &ReleaseLink{} (type *ReleaseLink) as type admission.Defaulter in assignment:
	*ReleaseLink does not implement admission.Defaulter (missing Default method)
# github.com/redhat-appstudio/release-service/api/v1alpha1
vet: api/v1alpha1/releaselink_webhook.go:38:27: cannot use &(ReleaseLink literal) (value of type *ReleaseLink) as admission.Defaulter value in variable declaration: missing method Default
make: *** [Makefile:107: vet] Error 2```

Reply:
Yeah, but is that because you created the webhook with the `--default` option and then you removed the function or it's always the case?

Reply:
I will double check

Reply:
When I do it that way, the `manifests.yaml` won't have a mutating webhook and I get ```Error: no matches for OriginalId admissionregistration.k8s.io_v1_MutatingWebhookConfiguration|~X|mutating-webhook-configuration; no matches for CurrentId admissionregistration.k8s.io_v1_MutatingWebhookConfiguration|~X|mutating-webhook-configuration; failed to find unique target for patch admissionregistration.k8s.io_v1_MutatingWebhookConfiguration|mutating-webhook-configuration
error: no objects passed to apply
make: *** [Makefile:149: deploy] Error 1``` when running `make deploy`

I am not guaranteeing there is no way to work around having that noop webhook there, but I haven't found a way so far